test/extended/router/gatewayapicontroller.go

```diff
@@ -0,0 +1,539 @@
+			csvName, err = oc.AsAdmin().Run("get").Args("-n", expectedSubscriptionNamespace, "subscription", expectedSubscriptionName, "-o=jsonpath={.status.installedCSV}").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			if csvName == "" {
+				e2e.Logf("Subscription %q doesn't have installed CSV, retrying...", expectedSubscriptionName)
+				return false, nil
+			}
+			e2e.Logf("Subscription %q has installed CSV: %s", expectedSubscriptionName, csvName)
+			return true, nil
+		})
+		o.Expect(waitVersionErr).NotTo(o.HaveOccurred(), "Timed out waiting for the ClusterServiceVersion to install")
+
+		waitCSVErr := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 2*time.Minute, false, func(context context.Context) (bool, error) {
+			csvStatus, err := oc.AsAdmin().Run("get").Args("-n", expectedSubscriptionNamespace, "clusterserviceversion", csvName, "-o=jsonpath={.status.phase}").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			if csvStatus != "Succeeded" {
+				e2e.Logf("Cluster Service Version %q is not successful, retrying...", csvName)
+				return false, nil
+			}
+			e2e.Logf("Cluster Service Version %q has succeeded!", csvName)
+			return true, nil
+		})
+		o.Expect(waitCSVErr).NotTo(o.HaveOccurred(), "Cluster Service Version %s never reached succeeded status", csvName)
+
+		// get OSSM Operator deployment
+		waitErr := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 2*time.Minute, false, func(context context.Context) (bool, error) {
+			deployOSSM, err := oc.AdminKubeClient().AppsV1().Deployments(expectedSubscriptionNamespace).Get(context, "servicemesh-operator3", metav1.GetOptions{})
+			if err != nil {
+				if deployOSSM.Status.ReadyReplicas < 1 {
+					e2e.Logf("OSSM operator deployment %q is not ready, retrying...", deploymentOSSMName)
+					return false, nil
+				}
+			}
+			e2e.Logf("OSSM operator deployment %q is ready", deploymentOSSMName)
+			return true, nil
+		})
+		o.Expect(waitErr).NotTo(o.HaveOccurred(), "OSSM Operator deployment %q did not successfully deploy its pod", deploymentOSSMName)
+
+		g.By("Confirm that Istio CR is created and in healthy state")
+		waitForIstioHealthy(oc)
+	})
+
+	g.It("Ensure default gatewayclass is accepted", func() {
+
+		g.By("Check if default GatewayClass is accepted after OLM resources are successful")
+		errCheck := checkGatewayClass(oc, gatewayClassName)
+		o.Expect(errCheck).NotTo(o.HaveOccurred(), "GatewayClass %q was not installed and accepted", gatewayClassName)
+
+	})
+
+	g.It("Ensure custom gatewayclass can be accepted", func() {
+		gwapiClient := gatewayapiclientset.NewForConfigOrDie(oc.AdminConfig())
+		customGatewayClassName := "gatewayclass-custom"
+
+		g.By("Ensure default GatewayClass is accepted and Istio is healthy")
+		errCheck := checkGatewayClass(oc, gatewayClassName)
+		o.Expect(errCheck).NotTo(o.HaveOccurred(), "GatewayClass %q was not installed and accepted", gatewayClassName)
+		waitForIstioHealthy(oc)
```

Comment:
If the test is the first one to be executed among the 5 tests, then we need to wait for Istio to be `Healthy` as well, otherwise it might be flaky in last checkpoint. Seems Istio status is not updated in time even gatewayclass status is accepted.